cmd/cnab-to-oci/push.go

```diff
@@ -38,6 +39,7 @@ func pushCmd() *cobra.Command {
 	cmd.Flags().StringVarP(&opts.targetRef, "target", "t", "", "reference where the bundle will be pushed")
 	cmd.Flags().StringSliceVar(&opts.insecureRegistries, "insecure-registries", nil, "Use plain HTTP for those registries")
 	cmd.Flags().BoolVar(&opts.allowFallbacks, "allow-fallbacks", true, "Enable automatic compatibility fallbacks for registries without support for custom media type, or OCI manifests")
+	cmd.Flags().StringVar(&opts.platform, "platform", "", "Single platform to push (instead of multi-arch images)")
```

Comment:
Perhaps it would be more flexible to instead allow for a list of platforms so users can pick a subset rather than forcing them into a 1 vs all decision?

Reply:
That modifies the algorithm quite a lot:
- currently I promote a single image in the list as the image to reference (a bit like when you do a `docker image pull`
- we would then need to dynamically generate manifest lists / oci indexes. If the manifest has any non-normated json fields (e.g., descriptor annotations in docker image lists), we will loose them.

Reply:
> we would then need to dynamically generate manifest lists / oci indexes.

We really can't just take the current one and filter the platforms field? That's a great pity.

Reply:
I think I have an idea that makes that possible. This involves parsing manifests in a weird way, but I'll give it a try.

Reply:
https://stackoverflow.com/questions/33436730/unmarshal-json-with-unknown-fields has some suggestions which might also be worth exploring. Decoding into a `map[string]json.RawMessage` looks especially promising I think.

Reply:
@ijc can you have a look at https://github.com/docker/cnab-to-oci/pull/33/commits/d08e835fb9fe7134a9b04b907bb60ed6877ecf4d and validate the approach ? (this is without adapting tests, nor toplevel command)

Reply:
Done. The approach seems valid to me. Only real comment is that there are many helpers in `container/platforms` for matching with lists of platforms etc which you've opencoded here.

Reply:
Thanks did not notice the platforms.Any function... Will use that